File: management/src/types/operationLog.ts
export type OperationLogValue = Record<string, unknown> | unknown[] | null

export interface OperationLogSummary {
  id: number
  admin_user_id: number | null
  admin_username: string
  module: string
  action: string
  created_at: string
}

export interface OperationLog extends OperationLogSummary {
  before_data: OperationLogValue
  after_data: OperationLogValue
}

export interface OperationLogFilters {
  module?: string
  q?: string
  limit?: number
}

export interface OperationLogBatchDeleteResult {
  deletedIds: number[]
  deletedCount: number
}

File: backend/app/schemas/operation_log.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class OperationLogListItemOut(BaseModel):
    id: int
    admin_user_id: int | None
    admin_username: str
    module: str
    action: str
    created_at: datetime

    model_config = {"from_attributes": True}


class OperationLogOut(OperationLogListItemOut):
    before_data: dict[str, Any] | list[Any] | None
    after_data: dict[str, Any] | list[Any] | None


class OperationLogBatchDeleteIn(BaseModel):
    ids: list[int] = Field(min_length=1, max_length=500)


class OperationLogBatchDeleteOut(BaseModel):
    deletedIds: list[int]
    deletedCount: int
